typescript/sdk/src/core/EvmCoreModule.ts

```diff
@@ -0,0 +1,251 @@
+      multiProvider,
+    );
+
+    // Deploy ICA ISM and Router
+    const { interchainAccountRouter, interchainAccountIsm } = (
+      await EvmIcaModule.create({
+        chain: chainName,
+        multiProvider,
+        config: {
+          mailbox: mailbox.address,
+          owner: await multiProvider.getSigner(chain).getAddress(),
+        },
+      })
+    ).serialize();
+
+    // Deploy Validator announce
+    const validatorAnnounce =
+      await hyperlaneCoreDeployer.deployValidatorAnnounce(
+        chainName,
+        mailbox.address,
+      );
+
+    // Deploy Test Receipient
+    const testRecipient = await hyperlaneCoreDeployer.deployTestRecipient(
+      chainName,
+      await mailbox.defaultIsm(),
+    );
+
+    return new EvmCoreModule(multiProvider, {
+      addresses: {
+        proxyAdmin,
+        mailbox,
+        validatorAnnounce,
+        interchainAccountIsm,
+        interchainAccountRouter,
+        testRecipient,
+        ismFactoryFactories: ismFactoriesFactory[chainName],
+      },
+      chain,
+      config,
+    });
+  }
+
+  static async deployProxyAdmin(
+    chainName: string,
+    hyperlaneCoreDeployer: HyperlaneCoreDeployer,
+  ) {
+    return hyperlaneCoreDeployer.deployContract(chainName, 'proxyAdmin', []);
+  }
+
+  static async deployIsmFactories(
+    chainName: string,
+    config: DerivedCoreConfig,
+    multiProvider: MultiProvider,
+  ): Promise<{
+    ismFactory: HyperlaneIsmFactory;
+    ismFactoriesFactory: ChainMap<HyperlaneContracts<ProxyFactoryFactories>>;
+  }> {
+    const proxyFactoryDeployer = new HyperlaneProxyFactoryDeployer(
+      multiProvider,
+    );
+    const ismFactoriesFactory = await proxyFactoryDeployer.deploy({
+      [chainName]: config,
+    });
+    const ismFactory = new HyperlaneIsmFactory(
+      ismFactoriesFactory,
+      multiProvider,
+    );
+
+    return { ismFactory, ismFactoriesFactory };
+  }
+
+  /**
+   * Deploys a Mailbox and it's default ISM, hook, and required hook contracts with a given configuration.
+   *
+   * @param chain - The chain name or ID to deploy the Mailbox on.
+   * @param config - The derived core configuration for the deployment.
+   * @param proxyAdmin - The address of the proxy admin for the Mailbox contract.
+   * @param deployer - The Hyperlane core deployer instance to use for the deployment.
+   * @param factories - The Hyperlane contract factories to use for the deployment.
+   * @param multiProvider - The multi-provider instance to use for the deployment.
+   * @returns The deployed Mailbox contract instance.
+   */
+  static async deployMailbox(
+    chain: ChainNameOrId,
+    config: DerivedCoreConfig,
+    proxyAdmin: Address,
+    deployer: HyperlaneCoreDeployer,
+    factories: HyperlaneContracts<ProxyFactoryFactories>,
+    multiProvider: MultiProvider,
+  ): Promise<Mailbox> {
+    const chainName = multiProvider.getChainName(chain);
+    const domain = deployer.multiProvider.getDomainId(chainName);
```

Comment:
why not just use the multiProvider you've passed in? also means you won't need to make it public in the core deployer

Reply:
fixed.